framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -3130,6 +3154,26 @@ protected void checkThrownExpression(ThrowTree tree) {
         }
     }
 
+    /**
+     * Returns a set of AnnotatedTypeMirror that is allowed throw exceptions for method.
+     *
+     * <p>Note: If the method declaration do not have any throw clauses, then return null.
+     *
+     * @param mtree MethodTree to look up throw expressions
+     * @return List of AnnotationTypeMirrors, one per exception in method signature, that contains
+     *     annotation and java expressions
+     */
+    protected List<AnnotatedTypeMirror> getExceptionList(MethodTree mtree) {
```

Comment:
Wouldn't this method be better in the `AnnotatedTypeFactory`? It's code that goes from a tree to annotated types, which is what the factory is for.